<a href="https://colab.research.google.com/github/Nahirmendozacrespo/analisis-movilidad-ecobici/blob/main/ecobici.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **ECOBICI** 🚴
Presento un analisis historico del desarrollo de las ecobicis (ciclovias)

Incorporación del primer conjunto de dataset-2024 para la exploración de columnas.



*   Usuarios ecobici
*   Recorridos realizados



In [2]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
ruta_archivo = '/content/drive/MyDrive/ecobici/usuarios-ecobici-2024.csv'
df_ecobici = pd.read_csv(ruta_archivo)
df_ecobici.head()

,id_usuario,genero_usuario,edad_usuario,fecha_alta,hora_alta
0,1094487,OTHER,35,2024-01-18,09:21:02
1,1097128,FEMALE,21,2024-01-21,23:02:26
2,1102901,OTHER,19,2024-01-29,19:33:32
3,1096100,FEMALE,29,2024-01-20,18:57:05
4,1103049,MALE,21,2024-01-30,00:49:33


Analisis exploratorio

In [ ]:
df_ecobici.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 197077 entries, 0 to 197076
Data columns (total 5 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   id_usuario      197077 non-null  int64 
 1   genero_usuario  197077 non-null  object
 2   edad_usuario    197077 non-null  int64 
 3   fecha_alta      197077 non-null  object
 4   hora_alta       197077 non-null  object
dtypes: int64(2), object(3)
memory usage: 7.5+ MB


In [ ]:
rutas = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2024.csv'
df_ruta = pd.read_csv(rutas)
df_ruta.head()

,id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,genero
0,20428222,568,2024-01-23 18:36:00,513,308 - SAN MARTIN II,Av. San Martín 5129,-58.490739,-34.597130,2024-01-23 18:45:28,498,055 - HABANA,Gral. José Gervasio Artigas 4298 (y Habana),-58.494959,-34.586598,992557.0,FIT,MALE
1,20431744,1355,2024-01-23 22:41:20,460,133 - BEIRO Y SEGUROLA,Segurola 3194,-58.511930,-34.607500,2024-01-23 23:03:55,382,204 - Biarritz,Biarritz 2403,-58.477255,-34.605431,320782.0,FIT,FEMALE
2,20429936,0,2024-01-23 20:06:22,467,328 - SARMIENTO II,Sarmiento 2037,-58.395893,-34.605514,2024-01-23 20:06:22,6,006 - Parque Lezama,"Avenida Martin Garcia, 295",-58.369758,-34.628526,828678.0,FIT,FEMALE
3,20429976,0,2024-01-23 20:08:17,382,204 - Biarritz,Biarritz 2403,-58.477255,-34.605431,2024-01-23 20:08:17,460,133 - BEIRO Y SEGUROLA,Segurola 3194,-58.511930,-34.607500,320782.0,ICONIC,FEMALE
4,20424802,680,2024-01-23 15:18:39,137,137 - AZOPARDO Y CHILE,AZOPARDO 700,-58.367492,-34.615598,2024-01-23 15:29:59,150,150 - RODRIGO BUENO,Av. España 2200,-58.355465,-34.618755,861425.0,FIT,FEMALE


In [ ]:
df_ruta.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3559284 entries, 0 to 3559283
Data columns (total 17 columns):
 #   Column                      Dtype  
---  ------                      -----  
 0   id_recorrido                int64  
 1   duracion_recorrido          int64  
 2   fecha_origen_recorrido      object 
 3   id_estacion_origen          int64  
 4   nombre_estacion_origen      object 
 5   direccion_estacion_origen   object 
 6   long_estacion_origen        float64
 7   lat_estacion_origen         float64
 8   fecha_destino_recorrido     object 
 9   id_estacion_destino         int64  
 10  nombre_estacion_destino     object 
 11  direccion_estacion_destino  object 
 12  long_estacion_destino       float64
 13  lat_estacion_destino        float64
 14  id_usuario                  float64
 15  modelo_bicicleta            object 
 16  genero                      object 
dtypes: float64(5), int64(4), object(8)
memory usage: 461.6+ MB


Concatenado de los dataset

> glob



**Concatenado bloque 1**


*   Ecobici-usuarios-2015
*   Ecobici-usuarios-2016
*   Ecobici-usuarios-2017
*   Ecobici-usuarios-2023
*   Ecobici-usuarios-2024
*   Ecobici-usuarios-2025

**Concatenado bloque 2**

*   Recorridos-realizados-2015
*   Recorridos-realizados-2016
*   Recorridos-realizados-2017
*   Recorridos-realizados-2023
*   Recorridos-realizados-2024
*   Recorridos-realizados-2025

**Concatenado bloque 3**

*   Recorridos-realizados-2010
*   Recorridos-realizados-2011
*   Recorridos-realizados-2012


In [ ]:
import pandas as pd
import glob

**Concatenado bloque 2**

In [ ]:
# 1. Define la ruta a la carpeta donde están TODOS tus CSVs.
# El "/*.csv" al final es clave: le dice a Python que busque todos los archivos con esa extensión.
# IMPORTANTE: Reemplaza TU_CARPETA con la ruta real.
ruta_carpeta = '/content/drive/MyDrive/ecobici/recorridos/*.csv'

# 2. Encontrar todos los archivos CSV en esa carpeta
archivos = glob.glob(ruta_carpeta)
print(f"Se encontraron {len(archivos)} archivos para unir.")

# 3. Crear una lista vacía para guardar temporalmente cada año
lista_dataframes = []

# 4. Leer cada archivo uno por uno y agregarlo a la lista
for archivo in archivos:
    # Leemos el archivo
    df_temporal = pd.read_csv(archivo)
    # Lo agregamos a la lista
    lista_dataframes.append(df_temporal)

# 5. Unir (concatenar) todos los años en un solo "súper" DataFrame
# ignore_index=True asegura que las filas se enumeren del 0 hasta el final sin repetir números
df_historico = pd.concat(lista_dataframes, ignore_index=True)

# 6. Ver el resultado final
print("Tamaño del dataset histórico (Filas, Columnas):")
print(df_historico.shape)

Se encontraron 6 archivos para unir.


/tmp/ipykernel_10925/3268401827.py:16: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.
  df_temporal = pd.read_csv(archivo)
/tmp/ipykernel_10925/3268401827.py:16: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.
  df_temporal = pd.read_csv(archivo)


Tamaño del dataset histórico (Filas, Columnas):
(11486014, 24)


In [ ]:
df_historico.head()

,periodo,genero_usuario,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,long_estacion_origen,lat_estacion_origen,domicilio_estacion_origen,duracion_recorrido,fecha_destino_recorrido,...,domicilio_estacion_destino,Unnamed: 0,Id_recorrido,direccion_estacion_origen,direccion_estacion_destino,id_usuario,modelo_bicicleta,género,id_recorrido,genero
0,2015.0,M,2015-05-01 00:00:18,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla,0 days 00:26:24.000000000,2015-05-01 00:26:42,...,Patricias Argentinas y Carlos Finlay,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,2015.0,F,2015-05-01 00:02:06,17,Plaza Almagro,-58.418832,-34.606399,Plaza Almagro,0 days 00:57:53.000000000,2015-05-01 00:59:59,...,Salguero Jerónimo y Mansilla,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,2015.0,M,2015-05-01 00:02:53,17,Plaza Almagro,-58.418832,-34.606399,Plaza Almagro,0 days 01:03:08.000000000,2015-05-01 01:06:01,...,Salguero Jerónimo y Mansilla,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,2015.0,M,2015-05-01 00:08:01,29,Parque Centenario,-58.434577,-34.608459,Patricias Argentinas y Carlos Finlay,0 days 01:04:18.000000000,2015-05-01 01:12:19,...,Salguero Jerónimo y Mansilla,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,2015.0,M,2015-05-01 00:08:57,29,Parque Centenario,-58.434577,-34.608459,Patricias Argentinas y Carlos Finlay,0 days 01:09:41.000000000,2015-05-01 01:18:38,...,Salguero Jerónimo y Mansilla,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


En el concatenado se visualizan datos NaN por falta de información. Que puede generarse porque con el paso del tiempo se recopila mayo información. Tenenemos una diferencia de columnas entre 2015-2025-

Vamos a explorar indivualmente los dataset.

In [ ]:
historico2015 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2015.csv'
df_historico2015 = pd.read_csv(historico2015)
df_historico2015.head()

,periodo,genero_usuario,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,long_estacion_origen,lat_estacion_origen,domicilio_estacion_origen,duracion_recorrido,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,long_estacion_destino,lat_estacion_destino,domicilio_estacion_destino
0,2015,M,2015-05-01 00:00:18,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla,0 days 00:26:24.000000000,2015-05-01 00:26:42,29,Parque Centenario,-58.434577,-34.608459,Patricias Argentinas y Carlos Finlay
1,2015,F,2015-05-01 00:02:06,17,Plaza Almagro,-58.418832,-34.606399,Plaza Almagro,0 days 00:57:53.000000000,2015-05-01 00:59:59,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla
2,2015,M,2015-05-01 00:02:53,17,Plaza Almagro,-58.418832,-34.606399,Plaza Almagro,0 days 01:03:08.000000000,2015-05-01 01:06:01,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla
3,2015,M,2015-05-01 00:08:01,29,Parque Centenario,-58.434577,-34.608459,Patricias Argentinas y Carlos Finlay,0 days 01:04:18.000000000,2015-05-01 01:12:19,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla
4,2015,M,2015-05-01 00:08:57,29,Parque Centenario,-58.434577,-34.608459,Patricias Argentinas y Carlos Finlay,0 days 01:09:41.000000000,2015-05-01 01:18:38,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla


In [ ]:
df_historico2015.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 503252 entries, 0 to 503251
Data columns (total 15 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   periodo                     503252 non-null  int64  
 1   genero_usuario              503252 non-null  object 
 2   fecha_origen_recorrido      503252 non-null  object 
 3   id_estacion_origen          503252 non-null  int64  
 4   nombre_estacion_origen      503252 non-null  object 
 5   long_estacion_origen        500084 non-null  float64
 6   lat_estacion_origen         500084 non-null  float64
 7   domicilio_estacion_origen   500084 non-null  object 
 8   duracion_recorrido          503252 non-null  object 
 9   fecha_destino_recorrido     503252 non-null  object 
 10  id_estacion_destino         503252 non-null  int64  
 11  nombre_estacion_destino     503252 non-null  object 
 12  long_estacion_destino       499790 non-null  float64
 13  lat_estacion_d

In [ ]:
df_ruta.info() #historico2024

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3559284 entries, 0 to 3559283
Data columns (total 17 columns):
 #   Column                      Dtype  
---  ------                      -----  
 0   id_recorrido                int64  
 1   duracion_recorrido          int64  
 2   fecha_origen_recorrido      object 
 3   id_estacion_origen          int64  
 4   nombre_estacion_origen      object 
 5   direccion_estacion_origen   object 
 6   long_estacion_origen        float64
 7   lat_estacion_origen         float64
 8   fecha_destino_recorrido     object 
 9   id_estacion_destino         int64  
 10  nombre_estacion_destino     object 
 11  direccion_estacion_destino  object 
 12  long_estacion_destino       float64
 13  lat_estacion_destino        float64
 14  id_usuario                  float64
 15  modelo_bicicleta            object 
 16  genero                      object 
dtypes: float64(5), int64(4), object(8)
memory usage: 461.6+ MB


In [ ]:
df_ruta.head() #historico2024

,id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,genero
0,20428222,568,2024-01-23 18:36:00,513,308 - SAN MARTIN II,Av. San Martín 5129,-58.490739,-34.597130,2024-01-23 18:45:28,498,055 - HABANA,Gral. José Gervasio Artigas 4298 (y Habana),-58.494959,-34.586598,992557.0,FIT,MALE
1,20431744,1355,2024-01-23 22:41:20,460,133 - BEIRO Y SEGUROLA,Segurola 3194,-58.511930,-34.607500,2024-01-23 23:03:55,382,204 - Biarritz,Biarritz 2403,-58.477255,-34.605431,320782.0,FIT,FEMALE
2,20429936,0,2024-01-23 20:06:22,467,328 - SARMIENTO II,Sarmiento 2037,-58.395893,-34.605514,2024-01-23 20:06:22,6,006 - Parque Lezama,"Avenida Martin Garcia, 295",-58.369758,-34.628526,828678.0,FIT,FEMALE
3,20429976,0,2024-01-23 20:08:17,382,204 - Biarritz,Biarritz 2403,-58.477255,-34.605431,2024-01-23 20:08:17,460,133 - BEIRO Y SEGUROLA,Segurola 3194,-58.511930,-34.607500,320782.0,ICONIC,FEMALE
4,20424802,680,2024-01-23 15:18:39,137,137 - AZOPARDO Y CHILE,AZOPARDO 700,-58.367492,-34.615598,2024-01-23 15:29:59,150,150 - RODRIGO BUENO,Av. España 2200,-58.355465,-34.618755,861425.0,FIT,FEMALE


In [ ]:
historico2017 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2017.csv'
df_historico2017 = pd.read_csv(historico2017)
df_historico2017.head()

/tmp/ipykernel_10925/2278496074.py:2: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.
  df_historico2017 = pd.read_csv(historico2017)


,periodo,genero_usuario,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,long_estacion_origen,lat_estacion_origen,domicilio_estacion_origen,duracion_recorrido,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,long_estacion_destino,lat_estacion_destino,domicilio_estacion_destino
0,2017,M,2017-01-01 00:02:14,60.0,25 de Mayo,-58.371155,-34.601638,25 de Mayo y Lavalle,0 days 00:27:44.000000000,2017-01-01 00:28:38,4.0,Plaza Roma,-58.368950,-34.601721,Lavalle y Bouchard
1,2017,M,2017-01-01 00:02:59,79.0,Azucena Villaflor,-58.363969,-34.611721,Azucena Villaflor y Pierina Dealessi,0 days 01:57:21.000000000,2017-01-01 01:00:52,54.0,Acuña de Figueroa,-58.421705,-34.598222,Lavalle y Acuña de Figueroa
2,2017,M,2017-01-01 00:07:13,11.0,Tribunales,-58.385022,-34.601313,Tribunales,0 days 00:25:08.000000000,2017-01-01 01:10:21,84.0,Lavalle,-58.395472,-34.602290,Ayacucho 595 entre Tucumán y Lavalle
3,2017,M,2017-01-01 00:15:12,79.0,Azucena Villaflor,-58.363969,-34.611721,Azucena Villaflor y Pierina Dealessi,0 days 00:33:39.000000000,2017-01-01 01:19:30,74.0,Instituto Leloir,-58.434243,-34.604572,"Av. Patricias Argentinas 401, entre Juana de I..."
4,2017,M,2017-01-01 00:15:27,150.0,Vera Peñaloza,-58.363091,-34.618082,Rosario Vera Peñaloza y Olga Cossettini,0 days 00:20:23.000000000,2017-01-01 01:25:08,1.0,Facultad de Derecho,-58.392452,-34.583133,Av. Pres.Figueroa Alcorta y Juan A.Bibiloni


In [ ]:
df_historico2017.info() #historico2017

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1048158 entries, 0 to 1048157
Data columns (total 15 columns):
 #   Column                      Non-Null Count    Dtype  
---  ------                      --------------    -----  
 0   periodo                     1048158 non-null  int64  
 1   genero_usuario              1048158 non-null  object 
 2   fecha_origen_recorrido      1048158 non-null  object 
 3   id_estacion_origen          1044739 non-null  float64
 4   nombre_estacion_origen      1048158 non-null  object 
 5   long_estacion_origen        1044739 non-null  float64
 6   lat_estacion_origen         1044739 non-null  float64
 7   domicilio_estacion_origen   1044739 non-null  object 
 8   duracion_recorrido          1048158 non-null  object 
 9   fecha_destino_recorrido     503252 non-null   object 
 10  id_estacion_destino         1044662 non-null  float64
 11  nombre_estacion_destino     1048158 non-null  object 
 12  long_estacion_destino       1044662 non-null  float64
 1

In [ ]:
historico2023 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2023.csv'
df_historico2023 = pd.read_csv(historico2023)
df_historico2023.head()

,Unnamed: 0,Id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,género
0,1,17910696BAEcobici,"1,848",2023-04-24 10:30:10,358BAEcobici,249 - Balbín,2519 Conesa,-58.465586,-34.561486,2023-04-24 11:00:58,278BAEcobici,233 - MONROE,2519 Superi,-58.469813,-34.564122,861866BAEcobici,ICONIC,MALE
1,2,17600256BAEcobici,288,2023-03-22 17:40:55,444BAEcobici,061 - Ministerio de Economia,"Balcarce e Yrigoyen, Hipolito Av.",-58.370716,-34.608936,2023-03-22 17:45:43,3BAEcobici,003 - ADUANA,Moreno & Azopardo,-58.368174,-34.611102,217525BAEcobici,ICONIC,FEMALE
2,3,17255670BAEcobici,"1,103",2023-02-15 23:12:22,280BAEcobici,222 - SIMON BOLIVAR,"1701 Fernandez Moreno, Baldomero",-58.449379,-34.633528,2023-02-15 23:30:45,280BAEcobici,222 - SIMON BOLIVAR,"1701 Fernandez Moreno, Baldomero",-58.449379,-34.633528,954201BAEcobici,ICONIC,MALE
3,4,17996972BAEcobici,"1,165",2023-05-03 11:00:20,273BAEcobici,223 - GAINZA,"494 Gainza, Martin De, Gral.",-58.446751,-34.616758,2023-05-03 11:19:45,367BAEcobici,287 - Belaustegui,"Belaustegui, Luis, Dr. 2890",-58.477209,-34.616212,179414BAEcobici,ICONIC,MALE
4,5,17148836BAEcobici,378,2023-02-06 06:50:58,65BAEcobici,065 - Julián Álvarez,3822 Guemes,-58.415787,-34.587312,2023-02-06 06:57:16,14BAEcobici,014 - Pacifico,"Santa Fe Av. & Bullrich, Int. Av.",-58.426387,-34.577424,8098BAEcobici,ICONIC,MALE


In [ ]:
df_historico2023.info() #historico2023

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2622331 entries, 0 to 2622330
Data columns (total 18 columns):
 #   Column                      Dtype  
---  ------                      -----  
 0   Unnamed: 0                  int64  
 1   Id_recorrido                object 
 2   duracion_recorrido          object 
 3   fecha_origen_recorrido      object 
 4   id_estacion_origen          object 
 5   nombre_estacion_origen      object 
 6   direccion_estacion_origen   object 
 7   long_estacion_origen        float64
 8   lat_estacion_origen         float64
 9   fecha_destino_recorrido     object 
 10  id_estacion_destino         object 
 11  nombre_estacion_destino     object 
 12  direccion_estacion_destino  object 
 13  long_estacion_destino       float64
 14  lat_estacion_destino        float64
 15  id_usuario                  object 
 16  modelo_bicicleta            object 
 17  género                      object 
dtypes: float64(4), int64(1), object(13)
memory usage: 360.1+ MB


In [ ]:
usuarios2023 = '/content/drive/MyDrive/ecobici/usuarios-ecobici-2023.csv'
df_usuarios2023 = pd.read_csv(usuarios2023)
df_usuarios2023.head()

/tmp/ipykernel_22170/2323419624.py:2: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_usuarios2023 = pd.read_csv(usuarios2023)


,ID_usuario,genero_usuario,edad_usuario,fecha_alta,hora_alta,Customer.Has.Dni..Yes...No.
0,1083476,MALE,20,2023-12-31,19:24:57,Yes
1,1083080,FEMALE,18,2023-12-31,00:05:03,Yes
2,1083163,OTHER,27,2023-12-31,10:16:12,No
3,1083435,MALE,24,2023-12-31,17:42:11,Yes
4,1083376,OTHER,32,2023-12-31,16:16:15,No


In [ ]:
df_usuarios2023.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 136066 entries, 0 to 136065
Data columns (total 6 columns):
 #   Column                       Non-Null Count   Dtype 
---  ------                       --------------   ----- 
 0   ID_usuario                   136066 non-null  int64 
 1   genero_usuario               136066 non-null  object
 2   edad_usuario                 136066 non-null  object
 3   fecha_alta                   136066 non-null  object
 4   hora_alta                    136066 non-null  object
 5   Customer.Has.Dni..Yes...No.  136066 non-null  object
dtypes: int64(1), object(5)
memory usage: 6.2+ MB


Verificamos que los datataset desde el 2015 al 2017 tiene menos columnas que los dataset del 2023 al 2025.



*   2015-2017 no tienen id_usuario y id_recorridos. No es posible usarlos como llave para unirlos en SQL.
*   2023-2025 tienen las mismas columnas con diferente formato. Por ejemplo: id_usuario en el dataset usuarios esta como int64 mientras que id_usuario en el dataset recorridos esta como object, tiene un sufijo luego del numero "EcoBici"

Vamos a borrar el sufijo de id_usuario para dejar solo los numeros y luego corroborar si son datos unicos que coinciden en ambos dataset.

Que sean datos unicos nos serviria para unir las tablas en SQL.



In [ ]:
# 1. Estandarizar el nombre de la columna para que se llamen EXACTAMENTE igual
df_usuarios2023 = df_usuarios2023.rename(columns={'ID_usuario': 'id_usuario'})

NameError: name 'df_usuarios2023' is not defined

In [ ]:
# 3. Limpiar la "basura" de la tabla de recorridos
df_historico2023['id_usuario'] = df_historico2023['id_usuario'].str.replace('BAEcobici', '', regex=False)

In [ ]:
# --- LA PRUEBA DE FUEGO DE LA LLAVE ---

# A. Verificamos que sea una "Llave Primaria" válida (sin duplicados)
usuarios_duplicados = df_usuarios2023['id_usuario'].duplicated().sum()
print(f"Usuarios duplicados en la tabla base: {usuarios_duplicados}")
if usuarios_duplicados == 0:
    print("¡Excelente! La columna funciona como una Llave Primaria perfecta.")
else:
    print("Cuidado: Hay IDs repetidos en la tabla de usuarios, habrá que limpiarlos.")

print("\n" + "-"*40 + "\n")

# B. El Momento de la Verdad: ¿Qué porcentaje de viajes cruzará con éxito?
viajes_con_match = df_historico2023['id_usuario'].isin(df_usuarios2023['id_usuario'])
porcentaje_exito = (viajes_con_match.mean() * 100)

print(f"Total de viajes en 2023: {len(df_historico2023):,}")
print(f"Viajes que SÍ encontraron a su usuario: {viajes_con_match.sum():,}")
print(f"Tasa de éxito del cruce (JOIN): {porcentaje_exito:.2f}%")

Usuarios duplicados en la tabla base: 0
¡Excelente! La columna funciona como una Llave Primaria perfecta.

----------------------------------------

Total de viajes en 2023: 2,622,331
Viajes que SÍ encontraron a su usuario: 637,480
Tasa de éxito del cruce (JOIN): 24.31%


Encontramos que hay recorridos sin id_usuario.

Vamos a crear una tabla con todos los usuarios desde 2015 hasta 2025.
Buscamos que todos los recorridos tengan su id_usuario.

# **Tabla usuarios 2015-2025**

In [ ]:
import pandas as pd
import glob

# 1. Definir la ruta específica para buscar solo los archivos de usuarios
# El asterisco (*) atrapará cualquier año en el nombre del archivo
# Asegúrate de que la ruta incluya tu carpeta 'ecobici' y el comodín '*.csv' al final
ruta_usuarios = '/content/drive/MyDrive/ecobici/usuarios-ecobici-*.csv'
archivos_usuarios = glob.glob(ruta_usuarios)
print(f"Se encontraron {len(archivos_usuarios)} archivos de usuarios para unir.")

lista_usuarios = []

for archivo in archivos_usuarios:
    df_temp = pd.read_csv(archivo)

    # 2. Estandarizar el nombre de la columna (por el problema de ID_usuario vs id_usuario)
    if 'ID_usuario' in df_temp.columns:
        df_temp = df_temp.rename(columns={'ID_usuario': 'id_usuario'})

    # 3. Forzar el ID a texto y limpiarlo de decimales (.0) o espacios
    if 'id_usuario' in df_temp.columns:
        df_temp['id_usuario'] = df_temp['id_usuario'].astype(str)
        df_temp['id_usuario'] = df_temp['id_usuario'].str.replace('.0', '', regex=False).str.strip()

    lista_usuarios.append(df_temp)

# 4. Concatenar toda la historia en una sola tabla gigante
df_usuarios_historico = pd.concat(lista_usuarios, ignore_index=True)
total_previo = len(df_usuarios_historico)

# 5. LA PRUEBA DE FUEGO: Eliminar usuarios duplicados
# subset=['id_usuario'] le dice a Pandas que busque repetidos basándose solo en el ID
# keep='last' asegura que si hay duplicados, nos quedamos con la versión más reciente de sus datos
df_usuarios_historico = df_usuarios_historico.drop_duplicates(subset=['id_usuario'], keep='last')
total_final = len(df_usuarios_historico)

print("\n--- RESUMEN DEL PADRÓN MAESTRO ---")
print(f"Registros brutos concatenados: {total_previo:,}")
print(f"Duplicados eliminados: {total_previo - total_final:,}")
print(f"🚴 Usuarios ÚNICOS finales: {total_final:,}")

Se encontraron 11 archivos de usuarios para unir.


/tmp/ipykernel_13626/3416820264.py:14: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  df_temp = pd.read_csv(archivo)



--- RESUMEN DEL PADRÓN MAESTRO ---
Registros brutos concatenados: 1,098,439
Duplicados eliminados: 220,395
🚴 Usuarios ÚNICOS finales: 878,044


In [ ]:
# 2. Convertir ambas a texto (nuestra regla de oro)
df_usuarios_historico['id_usuario'] = df_usuarios_historico['id_usuario'].astype(str)
df_historico2023['id_usuario'] = df_historico2023['id_usuario'].astype(str)

In [ ]:
# 3. Limpiar la "basura" de la tabla de recorridos
df_historico2023['id_usuario'] = df_historico2023['id_usuario'].str.replace('BAEcobici', '', regex=False)

In [ ]:
# 1. El Momento de la Verdad: ¿Qué porcentaje de viajes cruza con el Padrón Maestro?
# Usamos isin() para buscar cada ID de recorrido dentro del historial completo de usuarios
viajes_con_match_historico = df_historico2023['id_usuario'].isin(df_usuarios_historico['id_usuario'])

# 2. Calcular el porcentaje de éxito
porcentaje_exito_historico = (viajes_con_match_historico.mean() * 100)

# 3. Mostrar los resultados triunfales
print("--- PRUEBA DE FUEGO CON EL PADRÓN MAESTRO ---")
print(f"Total de viajes en 2023: {len(df_historico2023):,}")
print(f"Viajes que SÍ encontraron a su usuario: {viajes_con_match_historico.sum():,}")
print(f"Tasa de éxito del cruce (JOIN) actualizada: {porcentaje_exito_historico:.2f}%")

--- PRUEBA DE FUEGO CON EL PADRÓN MAESTRO ---
Total de viajes en 2023: 2,622,331
Viajes que SÍ encontraron a su usuario: 2,473,848
Tasa de éxito del cruce (JOIN) actualizada: 94.34%


Tenemos llave para unir las tablas en SQL con ID usuario entre tablas recorridos y usuarios. Tambien tenemos Padron historico del 2015-2025.

El cruce del padron con la tabla del 2023 "id_usuario" dio un 94.34% de confiabilidad.

Vamos a revisar el padron historico para limpiarlo y luego exportarlo.

In [ ]:
df_usuarios_historico.info()

<class 'pandas.core.frame.DataFrame'>
Index: 878044 entries, 0 to 1098438
Data columns (total 8 columns):
 #   Column                       Non-Null Count   Dtype  
---  ------                       --------------   -----  
 0   id_usuario                   878044 non-null  object 
 1   genero_usuario               877069 non-null  object 
 2   edad_usuario                 877277 non-null  object 
 3   fecha_alta                   878044 non-null  object 
 4   hora_alta                    878044 non-null  object 
 5   ID Cliente                   1 non-null       float64
 6   Customer Has Dni (Yes / No)  1 non-null       object 
 7   Customer.Has.Dni..Yes...No.  430806 non-null  object 
dtypes: float64(1), object(7)
memory usage: 60.3+ MB


In [ ]:
df_usuarios_historico.head()

,id_usuario,genero_usuario,edad_usuario,fecha_alta,hora_alta,ID Cliente,Customer Has Dni (Yes / No),Customer.Has.Dni..Yes...No.
0,7682,M,45,28-02-15,5:32:55 PM,NaN,NaN,NaN
1,19577,M,47,28-02-15,5:33:12 PM,NaN,NaN,NaN
3,129948,M,29,28-02-15,5:35:43 PM,NaN,NaN,NaN
4,129949,M,24,28-02-15,5:35:43 PM,NaN,NaN,NaN
5,129950,F,25,28-02-15,5:35:43 PM,NaN,NaN,NaN


Limpieza de columnas administrativas

In [ ]:
# 1. Definimos una lista con los nombres EXACTOS de las columnas a eliminar
columnas_basura = [
    'ID Cliente',
    'Customer Has Dni (Yes / No)',
    'Customer.Has.Dni..Yes...No.'
]

# 2. Le pedimos a Pandas que las elimine (drop) del Padrón Maestro
df_usuarios_historico = df_usuarios_historico.drop(columns=columnas_basura)

# 3. Volvemos a correr un info() rápido para confirmar que desaparecieron
print("--- RADIOGRAFÍA FINAL DEL PADRÓN LIMPIO ---")
df_usuarios_historico.info()

--- RADIOGRAFÍA FINAL DEL PADRÓN LIMPIO ---
<class 'pandas.core.frame.DataFrame'>
Index: 878044 entries, 0 to 1098438
Data columns (total 5 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   id_usuario      878044 non-null  object
 1   genero_usuario  877069 non-null  object
 2   edad_usuario    877277 non-null  object
 3   fecha_alta      878044 non-null  object
 4   hora_alta       878044 non-null  object
dtypes: object(5)
memory usage: 40.2+ MB


vamos a cambiar el tipo de datos de la columa fecha

In [ ]:
# Convertimos el texto a Fecha Real permitiendo formatos mezclados (mixed)
# dayfirst=True le da la pista a Pandas de que, en las fechas cortas, el primer número es el día
df_usuarios_historico['fecha_alta'] = pd.to_datetime(
    df_usuarios_historico['fecha_alta'],
    format='mixed',
    dayfirst=True
)

# Verificamos que ahora sí funcionó
print("Tipos de datos actualizados:")
print(df_usuarios_historico.dtypes)
print("\nPrimeras fechas modernizadas:")
print(df_usuarios_historico[['fecha_alta']].head())

Tipos de datos actualizados:
id_usuario                object
genero_usuario            object
edad_usuario              object
fecha_alta        datetime64[ns]
hora_alta                 object
dtype: object

Primeras fechas modernizadas:
  fecha_alta
0 2015-02-28
1 2015-02-28
3 2015-02-28
4 2015-02-28
5 2015-02-28


cambiamos tipo de dato de edad a float64

In [ ]:
# 1. Forzar la columna de edad a ser estrictamente numérica
df_usuarios_historico['edad_usuario'] = pd.to_numeric(df_usuarios_historico['edad_usuario'], errors='coerce')

# 2. Confirmar el cambio y ver estadísticas de la edad (promedio, mínimo y máximo)
print(df_usuarios_historico['edad_usuario'].describe())

count    877274.000000
mean         33.061183
std          14.585711
min        -976.000000
25%          24.000000
50%          30.000000
75%          40.000000
max        2021.000000
Name: edad_usuario, dtype: float64


In [ ]:
import numpy as np

In [ ]:
# 1. Definimos los límites de lo que consideramos una "edad humana lógica"
edad_minima = 16
edad_maxima = 100

# 2. Reemplazamos las edades fuera de este rango por NaN (Nulo)
df_usuarios_historico.loc[
    (df_usuarios_historico['edad_usuario'] < edad_minima) |
    (df_usuarios_historico['edad_usuario'] > edad_maxima),
    'edad_usuario'
] = np.nan

# 3. Volvemos a mirar las estadísticas para confirmar la limpieza
print("--- ESTADÍSTICAS DE EDAD LIMPIAS ---")
print(df_usuarios_historico['edad_usuario'].describe())

--- ESTADÍSTICAS DE EDAD LIMPIAS ---
count    877054.000000
mean         32.992381
std          11.776310
min          16.000000
25%          24.000000
50%          30.000000
75%          39.000000
max         100.000000
Name: edad_usuario, dtype: float64


Exportamos padron de usuarios del 2015-2025

In [ ]:
# Definimos la ruta de salida (asegúrate de que la carpeta 'ecobici' exista)
ruta_padron_final = '/content/drive/MyDrive/ecobici/usuarios_maestro_LIMPIO.csv'

# Exportamos el DataFrame a CSV sin la columna de índice
df_usuarios_historico.to_csv(ruta_padron_final, index=False)

print("✅ ¡Padrón Maestro de usuarios exportado y asegurado en Drive!")

✅ ¡Padrón Maestro de usuarios exportado y asegurado en Drive!


# **Dataset recorridos 2010-2025**


Bloque 1 para SQL 2023-2025

Bloque 2 historico 2010-2025



In [ ]:
recorridos2015 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2015.csv'
df_recorridos2015 = pd.read_csv(recorridos2015)
df_recorridos2015.head()

,periodo,genero_usuario,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,long_estacion_origen,lat_estacion_origen,domicilio_estacion_origen,duracion_recorrido,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,long_estacion_destino,lat_estacion_destino,domicilio_estacion_destino
0,2015,M,2015-05-01 00:00:18,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla,0 days 00:26:24.000000000,2015-05-01 00:26:42,29,Parque Centenario,-58.434577,-34.608459,Patricias Argentinas y Carlos Finlay
1,2015,F,2015-05-01 00:02:06,17,Plaza Almagro,-58.418832,-34.606399,Plaza Almagro,0 days 00:57:53.000000000,2015-05-01 00:59:59,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla
2,2015,M,2015-05-01 00:02:53,17,Plaza Almagro,-58.418832,-34.606399,Plaza Almagro,0 days 01:03:08.000000000,2015-05-01 01:06:01,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla
3,2015,M,2015-05-01 00:08:01,29,Parque Centenario,-58.434577,-34.608459,Patricias Argentinas y Carlos Finlay,0 days 01:04:18.000000000,2015-05-01 01:12:19,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla
4,2015,M,2015-05-01 00:08:57,29,Parque Centenario,-58.434577,-34.608459,Patricias Argentinas y Carlos Finlay,0 days 01:09:41.000000000,2015-05-01 01:18:38,25,Plaza Güemes,-58.416065,-34.589521,Salguero Jerónimo y Mansilla


In [ ]:
df_recorridos2015.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 503252 entries, 0 to 503251
Data columns (total 15 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   periodo                     503252 non-null  int64  
 1   genero_usuario              503252 non-null  object 
 2   fecha_origen_recorrido      503252 non-null  object 
 3   id_estacion_origen          503252 non-null  int64  
 4   nombre_estacion_origen      503252 non-null  object 
 5   long_estacion_origen        500084 non-null  float64
 6   lat_estacion_origen         500084 non-null  float64
 7   domicilio_estacion_origen   500084 non-null  object 
 8   duracion_recorrido          503252 non-null  object 
 9   fecha_destino_recorrido     503252 non-null  object 
 10  id_estacion_destino         503252 non-null  int64  
 11  nombre_estacion_destino     503252 non-null  object 
 12  long_estacion_destino       499790 non-null  float64
 13  lat_estacion_d

In [9]:
recorridos2025 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2025.csv'
df_recorridos2025 = pd.read_csv(recorridos2025)
df_recorridos2025.head()

,Id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,género
0,27699090BAEcobici,800,2025-10-15 11:12:52,146BAEcobici,146 - Hospital Francés,"3024 Calvo, Carlos",-58.407842,-34.622112,2025-10-15 11:26:12,17BAEcobici,017 - Plaza Almagro,"441 Bulnes & Peron, Juan Domingo, Tte. General",-58.418731,-34.606410,1177390BAEcobici,ICONIC,FEMALE
1,24622664BAEcobici,571,2025-01-10 08:48:58,70BAEcobici,070 - ARAOZ,1413 Araoz & Gorriti CABA,-58.426060,-34.592686,2025-01-10 08:58:29,74BAEcobici,074 - INSTITUTO LELOIR,Juana De Ibarbourou & Av Patricias Argentinas,-58.434540,-34.604390,357363BAEcobici,ICONIC,OTHER
2,24621373BAEcobici,348,2025-01-10 07:11:15,135BAEcobici,135 - MARCELO T. DE ALVEAR,Esmeralda & Paraguay,-58.378363,-34.597642,2025-01-10 07:17:03,130BAEcobici,130 - RETIRO II,Av. Dr. José María Ramos Mejía 1350,-58.374364,-34.591738,693254BAEcobici,ICONIC,MALE
3,27815450BAEcobici,343,2025-10-23 09:03:21,471BAEcobici,173 - EL LIBERTADOR,Av. San Martín 6220,-58.503309,-34.594729,2025-10-23 09:09:04,241BAEcobici,348 - Villa del Parque,"Gutierrez, Ricardo 3105",-58.494123,-34.600874,459726BAEcobici,ICONIC,MALE
4,28272276BAEcobici,931,2025-11-26 00:25:37,466BAEcobici,333 - PARQUE DE LA ESTACIÓN,Dr. Tomás Manuel de Anchorena 170,-58.411840,-34.608096,2025-11-26 00:41:08,527BAEcobici,068 - AV. CORDOBA,Av. Cordoba 5403,-58.438386,-34.588727,1153980BAEcobici,ICONIC,MALE


In [10]:
recorridos2024 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2024.csv'
df_recorridos2024 = pd.read_csv(recorridos2024)
df_recorridos2024.head()

,id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,genero
0,20428222,568,2024-01-23 18:36:00,513,308 - SAN MARTIN II,Av. San Martín 5129,-58.490739,-34.597130,2024-01-23 18:45:28,498,055 - HABANA,Gral. José Gervasio Artigas 4298 (y Habana),-58.494959,-34.586598,992557.0,FIT,MALE
1,20431744,1355,2024-01-23 22:41:20,460,133 - BEIRO Y SEGUROLA,Segurola 3194,-58.511930,-34.607500,2024-01-23 23:03:55,382,204 - Biarritz,Biarritz 2403,-58.477255,-34.605431,320782.0,FIT,FEMALE
2,20429936,0,2024-01-23 20:06:22,467,328 - SARMIENTO II,Sarmiento 2037,-58.395893,-34.605514,2024-01-23 20:06:22,6,006 - Parque Lezama,"Avenida Martin Garcia, 295",-58.369758,-34.628526,828678.0,FIT,FEMALE
3,20429976,0,2024-01-23 20:08:17,382,204 - Biarritz,Biarritz 2403,-58.477255,-34.605431,2024-01-23 20:08:17,460,133 - BEIRO Y SEGUROLA,Segurola 3194,-58.511930,-34.607500,320782.0,ICONIC,FEMALE
4,20424802,680,2024-01-23 15:18:39,137,137 - AZOPARDO Y CHILE,AZOPARDO 700,-58.367492,-34.615598,2024-01-23 15:29:59,150,150 - RODRIGO BUENO,Av. España 2200,-58.355465,-34.618755,861425.0,FIT,FEMALE


In [11]:
recorridos2023 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2023.csv'
df_recorridos2023 = pd.read_csv(recorridos2023)
df_recorridos2023.head()

,Unnamed: 0,Id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,género
0,1,17910696BAEcobici,"1,848",2023-04-24 10:30:10,358BAEcobici,249 - Balbín,2519 Conesa,-58.465586,-34.561486,2023-04-24 11:00:58,278BAEcobici,233 - MONROE,2519 Superi,-58.469813,-34.564122,861866BAEcobici,ICONIC,MALE
1,2,17600256BAEcobici,288,2023-03-22 17:40:55,444BAEcobici,061 - Ministerio de Economia,"Balcarce e Yrigoyen, Hipolito Av.",-58.370716,-34.608936,2023-03-22 17:45:43,3BAEcobici,003 - ADUANA,Moreno & Azopardo,-58.368174,-34.611102,217525BAEcobici,ICONIC,FEMALE
2,3,17255670BAEcobici,"1,103",2023-02-15 23:12:22,280BAEcobici,222 - SIMON BOLIVAR,"1701 Fernandez Moreno, Baldomero",-58.449379,-34.633528,2023-02-15 23:30:45,280BAEcobici,222 - SIMON BOLIVAR,"1701 Fernandez Moreno, Baldomero",-58.449379,-34.633528,954201BAEcobici,ICONIC,MALE
3,4,17996972BAEcobici,"1,165",2023-05-03 11:00:20,273BAEcobici,223 - GAINZA,"494 Gainza, Martin De, Gral.",-58.446751,-34.616758,2023-05-03 11:19:45,367BAEcobici,287 - Belaustegui,"Belaustegui, Luis, Dr. 2890",-58.477209,-34.616212,179414BAEcobici,ICONIC,MALE
4,5,17148836BAEcobici,378,2023-02-06 06:50:58,65BAEcobici,065 - Julián Álvarez,3822 Guemes,-58.415787,-34.587312,2023-02-06 06:57:16,14BAEcobici,014 - Pacifico,"Santa Fe Av. & Bullrich, Int. Av.",-58.426387,-34.577424,8098BAEcobici,ICONIC,MALE


In [12]:
recorridos2022 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2022.csv'
df_recorridos2022 = pd.read_csv(recorridos2022)
df_recorridos2022.head()

,Unnamed: 0,X,Id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,Género
0,1,1,13267975BAEcobici,"2,610",2022-01-16 14:58:42,5BAEcobici,005 - Plaza Italia,Av. Sarmiento 2601,-58.420954,-34.580550,2022-01-16 15:42:12,210BAEcobici,335 - General Urquiza,Figueroa Alcorta & Sarmiento,-58.411278,-34.572165,776361BAEcobici,ICONIC,FEMALE
1,2,2,13268526BAEcobici,545,2022-01-16 17:26:27,51BAEcobici,051 - TUCUMAN,Tucuman & 9 De Julio Av.,-58.382126,-34.601478,2022-01-16 17:35:32,174BAEcobici,174 - MINISTERIO DE EDUCACION,Marcelo T. de Alvear & Rodriguez Peña,-58.391768,-34.597225,776407BAEcobici,ICONIC,MALE
2,3,3,13268400BAEcobici,"2,061",2022-01-16 16:51:12,161BAEcobici,161 - Humahuaca,3912 Humahuaca,-58.419676,-34.602078,2022-01-16 17:25:33,117BAEcobici,117 - HUMBERTO 1°,Peru 1016,-58.374176,-34.620101,671762BAEcobici,ICONIC,FEMALE
3,4,4,13268164BAEcobici,"12,748",2022-01-16 15:58:01,210BAEcobici,335 - General Urquiza,Figueroa Alcorta & Sarmiento,-58.411278,-34.572165,2022-01-16 19:30:29,382BAEcobici,204 - Biarritz,Biarritz 2403,-58.477390,-34.605470,776361BAEcobici,ICONIC,FEMALE
4,5,5,13270010BAEcobici,"4,337",2022-01-16 23:40:09,215BAEcobici,113 - Guatemala,Guatemala 4773,-58.424996,-34.585878,2022-01-17 00:52:26,205BAEcobici,125 - F.J.Santamaria de Oro,F.J.Santamaria de Oro & Guatemala,-58.428016,-34.583323,454615BAEcobici,ICONIC,OTHER


In [13]:
recorridos2021 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2021.csv'
df_recorridos2021 = pd.read_csv(recorridos2021)
df_recorridos2021.head()

/tmp/ipykernel_41911/658294850.py:2: DtypeWarning: Columns (17,18) have mixed types. Specify dtype option on import or set low_memory=False.
  df_recorridos2021 = pd.read_csv(recorridos2021)


,Unnamed: 0,Id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,género,Género
0,1,9920899BAEcobici,"2,026",2021-01-26 19:58:12,86BAEcobici,086 - SAAVEDRA,1019 Saavedra,-58.401681,-34.621268,2021-01-26 20:31:58,8BAEcobici,008 - Congreso,"Cevallos, Virrey& Yrigoyen, Hipolito Av.",-34.609422,"-34.6094218,-58.3893364",289347BAEcobici,ICONIC,OTHER,NaN
1,2,9918792BAEcobici,"1,474",2021-01-26 17:37:33,197BAEcobici,197 - PLAZA CIUDAD DE UDINE,Mercedes 1480,-58.493044,-34.620998,2021-01-26 18:02:07,9BAEcobici,009 - Parque Las Heras,"2300 Diaz, Cnel. Av. & French",-34.585443,"-34.585443,-58.407741",637104BAEcobici,ICONIC,OTHER,NaN
2,3,9917121BAEcobici,629,2021-01-26 15:24:59,165BAEcobici,165 - PLAZA MONSEÑOR MIGUEL DE ANDREA,Dr Tomas Manuel De Anchorena & Jose A. Cabrera,-58.407614,-34.597048,2021-01-26 15:35:28,13BAEcobici,013 - ONCE,Av. Pueyrredon & Av. Rivadavia,-34.610090,"-34.61009,-58.406",709319BAEcobici,ICONIC,FEMALE,NaN
3,4,9919277BAEcobici,616,2021-01-26 18:11:22,20BAEcobici,020 - DISTRITO AUDIOVISUAL,Conde & Concepción Arenal,-58.444327,-34.580975,2021-01-26 18:21:38,14BAEcobici,014 - Pacifico,"Santa Fe Av. & Bullrich, Int. Av.",-34.577424,"-34.577424,-58.426387",114280BAEcobici,ICONIC,FEMALE,NaN
4,5,9921579BAEcobici,684,2021-01-26 20:38:30,44BAEcobici,044 - Ecoparque,"Av. Del Libertador, 3260",-58.413883,-34.575515,2021-01-26 20:49:54,14BAEcobici,014 - Pacifico,"Santa Fe Av. & Bullrich, Int. Av.",-34.577424,"-34.577424,-58.426387",641027BAEcobici,ICONIC,FEMALE,NaN


In [14]:
recorridos2020 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2020.csv'
df_recorridos2020 = pd.read_csv(recorridos2020)
df_recorridos2020.head()

,Unnamed: 0,Id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,género
0,1,7210548BAEcobici,"1,582",2020-01-24 21:54:39,27BAEcobici,027 - Montevideo,Cordoba Av. & Montevideo,-58.390089,-34.599068,2020-01-24 22:21:01,3BAEcobici,003 - ADUANA,Moreno & Av Paseo Colon,-34.611032,"-34.611032,-58.3682604",192009BAEcobici,ICONIC,FEMALE
1,2,7199093BAEcobici,204,2020-01-24 07:04:19,151BAEcobici,151 - AIME PAINÉ,"Villaflor, Azucena & Paine, Aime",-58.361285,-34.611815,2020-01-24 07:07:43,3BAEcobici,003 - ADUANA,Moreno & Av Paseo Colon,-34.611032,"-34.611032,-58.3682604",36380BAEcobici,ICONIC,MALE
2,3,7196805BAEcobici,"1,790",2020-01-24 00:15:17,111BAEcobici,111 - MACACHA GUEMES,Machaca Guemes 350,-58.364686,-34.605488,2020-01-24 00:45:07,3BAEcobici,003 - ADUANA,Moreno & Av Paseo Colon,-34.611032,"-34.611032,-58.3682604",460080BAEcobici,ICONIC,MALE
3,4,7203598BAEcobici,"10,688",2020-01-24 12:38:16,285BAEcobici,400 - Reserva Ecologica,"Achaval Rodriguez, T., Dr. Av. 1520",-58.356175,-34.617212,2020-01-24 15:36:24,4BAEcobici,004 - Plaza Roma,Lavalle & Bouchard,-34.601822,"-34.601822,-58.368781",3857BAEcobici,ICONIC,MALE
4,5,7200335BAEcobici,673,2020-01-24 08:31:01,171BAEcobici,171 - Pasteur,519 Pasteur,-58.399755,-34.603281,2020-01-24 08:42:14,7BAEcobici,007 - OBELISCO,CARLOS PELEGRINI 215,-34.606498,"-34.606498,-58.381098",391034BAEcobici,ICONIC,FEMALE


In [3]:
recorridos2019 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2019.csv'
df_recorridos2019 = pd.read_csv(recorridos2019)
df_recorridos2019.head()

,Unnamed: 0,Id_recorrido,duracion_recorrido,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,direccion_estacion_origen,long_estacion_origen,lat_estacion_origen,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,direccion_estacion_destino,long_estacion_destino,lat_estacion_destino,id_usuario,modelo_bicicleta,género
0,1,9379BAEcobici,239,2019-02-27 19:44:42,5BAEcobici,005 - Plaza Italia,Av. Sarmiento 2601,-58.420954,-34.580550,2019-02-27 19:48:41,5BAEcobici,005 - Plaza Italia,Av. Sarmiento 2601,-34.580550,"-34.5805497,-58.4209542",6243BAEcobici,ICONIC,OTHER
1,2,7441BAEcobici,"2,814",2019-02-27 14:35:58,6BAEcobici,006 - Parque Lezama,"Avenida Martin Garcia, 295",-58.369758,-34.628526,2019-02-27 15:22:52,6BAEcobici,006 - Parque Lezama,"Avenida Martin Garcia, 295",-34.628526,"-34.628526,-58.369758",11140BAEcobici,ICONIC,MALE
2,3,6828BAEcobici,298,2019-02-27 11:51:40,73BAEcobici,073 - Ruy Díaz de Guzmán,Avenida Martin Garcia y Ruy Díaz de Guzmán,-58.371823,-34.630681,2019-02-27 11:56:38,6BAEcobici,006 - Parque Lezama,"Avenida Martin Garcia, 295",-34.628526,"-34.628526,-58.369758",1994BAEcobici,ICONIC,MALE
3,4,7485BAEcobici,"2,324",2019-02-27 14:45:57,59BAEcobici,018 - Independencia,Idependencia y Bernardo de Irigoyen,-58.380565,-34.617654,2019-02-27 15:24:41,9BAEcobici,009 - Parque Las Heras,"2300 Diaz, Cnel. Av. & French",-34.585443,"-34.585443,-58.407741",2176BAEcobici,ICONIC,MALE
4,5,6111BAEcobici,"1,215",2019-02-27 07:46:13,91BAEcobici,091 - Pasco,708 Pasco & Chile,-58.397602,-34.617448,2019-02-27 08:06:28,10BAEcobici,KP1808000368,"1600 Moreau De Justo, Alicia Av.",-34.616282,"-34.6162819,-58.3657444",6056BAEcobici,ICONIC,FEMALE


In [4]:
recorridos2018 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2018.csv'
df_recorridos2018 = pd.read_csv(recorridos2018)
df_recorridos2018.head()

,periodo,id_usuario,genero_usuario,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,long_estacion_origen,lat_estacion_origen,domicilio_estacion_origen,duracion_recorrido,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,long_estacion_destino,lat_estacion_destino,domicilio_estacion_destino
0,2018,5453,M,2018-01-01 00:08:05,45.0,Uruguay,-58.386913,-34.601616,Tucumán y Uruguay,0 days 00:19:53.000000000,2018-01-01 00:27:58,183.0,Virrey Cevallos,-58.389692,-34.615638,México 1652 entre Virrey Cevallos y Solís
1,2018,673,M,2018-01-01 00:18:05,189.0,Posadas,-58.385586,-34.588475,Posadas 1350 entre Montevideo y Rodríguez Peña,0 days 00:26:19.000000000,2018-01-01 00:44:24,110.0,Guardia Vieja,-58.415459,-34.601761,Guardia Vieja 3606 entre Billinghurst y Mario ...
2,2018,179119,F,2018-01-01 00:20:14,50.0,Hospital Rivadavia,-58.401148,-34.583677,Lucena Pereyra 2516 entre Las Heras y Pagano,0 days 00:27:39.000000000,2018-01-01 00:47:53,31.0,Padilla,-58.439709,-34.603162,Padilla y Warnes Av
3,2018,400147,M,2018-01-01 00:20:22,111.0,Macacha Güemes,-58.363000,-34.604969,Macacha Guemes y Juana Manso,0 days 00:48:51.000000000,2018-01-01 01:09:13,54.0,Acuña de Figueroa,-58.421705,-34.598222,Lavalle y Acuña de Figueroa
4,2018,400156,F,2018-01-01 00:20:31,111.0,Macacha Güemes,-58.363000,-34.604969,Macacha Guemes y Juana Manso,0 days 00:49:27.000000000,2018-01-01 01:09:58,54.0,Acuña de Figueroa,-58.421705,-34.598222,Lavalle y Acuña de Figueroa


In [5]:
recorridos2017 = '/content/drive/MyDrive/ecobici/recorridos/recorridos-realizados-2017.csv'
df_recorridos2017 = pd.read_csv(recorridos2017)
df_recorridos2017.head()

/tmp/ipykernel_41911/3315251177.py:2: DtypeWarning: Columns (9) have mixed types. Specify dtype option on import or set low_memory=False.
  df_recorridos2017 = pd.read_csv(recorridos2017)


,periodo,genero_usuario,fecha_origen_recorrido,id_estacion_origen,nombre_estacion_origen,long_estacion_origen,lat_estacion_origen,domicilio_estacion_origen,duracion_recorrido,fecha_destino_recorrido,id_estacion_destino,nombre_estacion_destino,long_estacion_destino,lat_estacion_destino,domicilio_estacion_destino
0,2017,M,2017-01-01 00:02:14,60.0,25 de Mayo,-58.371155,-34.601638,25 de Mayo y Lavalle,0 days 00:27:44.000000000,2017-01-01 00:28:38,4.0,Plaza Roma,-58.368950,-34.601721,Lavalle y Bouchard
1,2017,M,2017-01-01 00:02:59,79.0,Azucena Villaflor,-58.363969,-34.611721,Azucena Villaflor y Pierina Dealessi,0 days 01:57:21.000000000,2017-01-01 01:00:52,54.0,Acuña de Figueroa,-58.421705,-34.598222,Lavalle y Acuña de Figueroa
2,2017,M,2017-01-01 00:07:13,11.0,Tribunales,-58.385022,-34.601313,Tribunales,0 days 00:25:08.000000000,2017-01-01 01:10:21,84.0,Lavalle,-58.395472,-34.602290,Ayacucho 595 entre Tucumán y Lavalle
3,2017,M,2017-01-01 00:15:12,79.0,Azucena Villaflor,-58.363969,-34.611721,Azucena Villaflor y Pierina Dealessi,0 days 00:33:39.000000000,2017-01-01 01:19:30,74.0,Instituto Leloir,-58.434243,-34.604572,"Av. Patricias Argentinas 401, entre Juana de I..."
4,2017,M,2017-01-01 00:15:27,150.0,Vera Peñaloza,-58.363091,-34.618082,Rosario Vera Peñaloza y Olga Cossettini,0 days 00:20:23.000000000,2017-01-01 01:25:08,1.0,Facultad de Derecho,-58.392452,-34.583133,Av. Pres.Figueroa Alcorta y Juan A.Bibiloni


In [15]:
df_recorridos2025.info()
df_recorridos2024.info()
df_recorridos2023.info()
df_recorridos2022.info()
df_recorridos2021.info()
df_recorridos2020.info()
df_recorridos2019.info()
df_recorridos2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3156182 entries, 0 to 3156181
Data columns (total 17 columns):
 #   Column                      Dtype  
---  ------                      -----  
 0   Id_recorrido                object 
 1   duracion_recorrido          object 
 2   fecha_origen_recorrido      object 
 3   id_estacion_origen          object 
 4   nombre_estacion_origen      object 
 5   direccion_estacion_origen   object 
 6   long_estacion_origen        float64
 7   lat_estacion_origen         float64
 8   fecha_destino_recorrido     object 
 9   id_estacion_destino         object 
 10  nombre_estacion_destino     object 
 11  direccion_estacion_destino  object 
 12  long_estacion_destino       float64
 13  lat_estacion_destino        float64
 14  id_usuario                  object 
 15  modelo_bicicleta            object 
 16  género                      object 
dtypes: float64(4), object(13)
memory usage: 409.4+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 355